tests/posix_semaphore/main.c

```diff
@@ -236,16 +236,16 @@ void test3(void)
     sem_post(&s1);
 }
 
-#ifdef BOARD_NATIVE
-/* native can sometime take more time to respond as it is not real time */
-#define TEST4_TIMEOUT_EXCEEDED_MARGIN (300)
-#elif CPU_FAM_NRF51
-/* nrf51 based boards needs a slightly higher margin value. Using 105us makes
- test4 result more reliable. */
-#define TEST4_TIMEOUT_EXCEEDED_MARGIN (105)
-#else
-#define TEST4_TIMEOUT_EXCEEDED_MARGIN (100)
-#endif /* BOARD_NATIVE */
+/*
+ * Allowed margin for waiting too long.
+ *
+ * Waiting too short is forbidden by POSIX, but is checked elsewhere.
```

Comment:
It is checked here, not elsewhere:

```
    if (elapsed < exp) {
        printf("first: waited only %s usec => FAILED\n", uint64_str);
    }
```

Reply:
What it is it?

Reply:
The check in the code that verifies it does not wait too short.